unit_tests/unit/test_log_archive.py

```
@@ -247,3 +247,79 @@ def test_parallel_compression_produces_valid_archives(tmp_path, large_log_file):
+        '                out_name="${timestamp}.${chunk_seq}.${ARCHIVE_NAME}.zst"\n'
+        "            fi\n"
+        '            { printf "%s\\n" "$first_line"; cat; } | zstd -q -o "$out_name"\n'
+        "        fi\n"
+        '    \' "$INPUT_FILE" "$WORK_DIR/chunk_"\n'
+        "kill $SAMPLER_PID 2>/dev/null || true\n"
+        "wait $SAMPLER_PID 2>/dev/null || true\n"
+    )
+    wrapper.chmod(0o755)
+
+    max_size_file = tmp_path / "max_work_dir_size"
+    result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)